examples/official-site/sqlpage/migrations/33_carousel_component.sql

```diff
@@ -0,0 +1,138 @@
+INSERT INTO component (name, description, icon, introduced_in_version)
+VALUES (
+        'carousel',
+        'A carousel is used to display multiple pieces of visual content without taking up too much space.',
+        'carousel-horizontal',
+        '0.18.0'
+    );
+
+INSERT INTO parameter (
+        component,
+        name,
+        description,
+        type,
+        top_level,
+        optional
+    )
+VALUES (
+        'carousel',
+        'name',
+        'An unique string to identify the caroussel component in the HTML page.',
+        'TEXT',
+        TRUE,
+        FALSE
```

Comment:
We try to avoid having such requirements in sqlpage. We can do with it if there is no other choice, but it looks like this is an internal implementation technicality. Wouldn't it be better if the user didn't have to specify an id and make sure it is unique ? It is very easy to end up with duplicate identifiers and errors that are completely impossible to debug if you don't already understand html, javascript and the DOM.

Couldn't we auto-generate an id like `carousel_{{query_number}}` where query_number comes from https://github.com/lovasoa/SQLpage/blob/main/src/render.rs ?

Reply:
Yes, it's possible to use an id like carousel_{{query_number}} but it seems to me not accessible in the template. If I understand the Rust code, the value of query_number is only initialized where an error is occured and only in development mode.

Reply:
Yes render.rs needs to be updated to add query_number to the handlebars context.

Reply:
You can look at how `row_index` is added to the context.
`query_number` needs to be passed all the way down to the `SplitTemplateRenderer`